pkg/sql/catalog/randgen/batch.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package randgen
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+)
+
+// newDesc queues a freshly created descriptor for (later) write to KV.
+// It ensures a namespace entry will also be created.
+func (g *testSchemaGenerator) newDesc(ctx context.Context, desc catalog.MutableDescriptor) {
+	if g.cfg.DryRun {
+		return
+	}
+	g.newDescs = append(g.newDescs, desc)
+	g.queueDescMut(ctx, desc)
+}
+
+// queueDesc queues a modified descriptor for (later) write to KV.
+func (g *testSchemaGenerator) queueDescMut(ctx context.Context, desc catalog.MutableDescriptor) {
```

Comment:
I think it's worth calling `validate.Self(clusterversion.TestingClusterVersion, desc)` right here. Validation will be performed anyway when the transaction commits but since a lot of descriptors are involved, _if_ a validation failure occurs it might be difficult to make sense of it.